it/src/test/scala/TransactionGenerator/TransactionsUtil.scala

```diff
@@ -0,0 +1,182 @@
+                                      contract: CompiledContract,
+                                      numberOfTokensForIssue: Long,
+                                      tokenIdOpt: Option[ADKey] = None): Transaction = {
+    val directiveForChange: TransferDirective =
+      createDirectiveForChange(privKey, fee, amount = 0, useOutputs.map(_._1.amount).sum)
+    val directiveForScriptedAssetIssue: IndexedSeq[Directive] =
+      IndexedSeq(ScriptedAssetDirective(contract.hash, numberOfTokensForIssue, tokenIdOpt)) :+ directiveForChange
+    prepareTransaction(privKey, fee, timestamp, useOutputs, directiveForScriptedAssetIssue)
+  }
+
+  def assetIssuingTransactionScratch(privKey: PrivateKey25519,
+                                     fee: Long,
+                                     timestamp: Long,
+                                     useOutputs: Seq[(MonetaryBox, Option[(CompiledContract, Seq[Proof])])],
+                                     contract: CompiledContract,
+                                     numberOfTokensForIssue: Long): Transaction = {
+    val directiveForChange: TransferDirective =
+      createDirectiveForChange(privKey, fee, amount = 0, useOutputs.map(_._1.amount).sum)
+    val directiveForTokenIssue: IndexedSeq[Directive] =
+      IndexedSeq(AssetIssuingDirective(contract.hash, numberOfTokensForIssue)) :+ directiveForChange
+    prepareTransaction(privKey, fee, timestamp, useOutputs, directiveForTokenIssue)
+  }
+
+  def dataTransactionScratch(privKey: PrivateKey25519,
+                             fee: Long,
+                             timestamp: Long,
+                             useOutputs: Seq[(MonetaryBox, Option[(CompiledContract, Seq[Proof])])],
+                             contract: CompiledContract,
+                             data: Array[Byte]): Transaction = {
+    val directiveForChange: TransferDirective =
+      createDirectiveForChange(privKey, fee, amount = 0, useOutputs.map(_._1.amount).sum)
+    val directiveForDataTransaction: IndexedSeq[Directive] =
+      IndexedSeq(DataDirective(contract.hash, data)) :+ directiveForChange
+    prepareTransaction(privKey, fee, timestamp, useOutputs, directiveForDataTransaction)
+  }
+
+  private def createDirectiveForChange(privKey: PrivateKey25519,
+                                       fee: Long,
+                                       amount: Long,
+                                       totalAmount: Long,
+                                       tokenId: Option[TokenId] = None): TransferDirective = {
+    val change: Long = totalAmount - (amount + fee)
+    if (change < 0) {
+      logger.warn(s"Transaction impossible: required amount is bigger than available. Change is: $change.")
+      throw new RuntimeException(s"Transaction impossible: required amount is bigger than available $change")
+    }
+    TransferDirective(privKey.publicImage.address.address, change, tokenId.map(element => ADKey @@ element))
```

Comment:
You should also check the condition ("change == 0"), otherwise it is possible to generate directive with amount = 0